code/controllers/subsystem/achievements.dm

```diff
@@ -101,3 +101,33 @@ SUBSYSTEM_DEF(achievements)
 
 	if(to_update.len)
 		SSdbcore.MassInsert(format_table_name("achievement_metadata"),to_update,duplicate_key = TRUE)
+
+	var/list/orphaned_keys = get_orphaned_keys()
+	if(orphaned_keys.len)
+		message_admins("Achievement metadata found without matching achievement, use Achievement-Admin-Panel verb to cleanup if necessary")
```

Comment:
Thoughts on a config to suppress this message in case server ops get frustrated with achievements they don't want to delete (for some odd reason). Default can be sending the message.